mapzen-places-api/src/test/java/com/mapzen/places/api/internal/PeliasPlaceDetailFetcherTest.java

```diff
@@ -1,44 +1,60 @@
 package com.mapzen.places.api.internal;
 
 import com.mapzen.pelias.Pelias;
-import com.mapzen.tangram.LngLat;
 
 import org.junit.Before;
 import org.junit.Test;
 
 import java.util.HashMap;
 import java.util.Map;
 
+import static com.mapzen.places.api.internal.PlaceDetailFetcher.PELIAS_GID_BASE;
+import static com.mapzen.places.api.internal.PlaceDetailFetcher.PELIAS_GID_NODE;
+import static com.mapzen.places.api.internal.PlaceDetailFetcher.PELIAS_GID_WAY;
+import static com.mapzen.places.api.internal.PlaceDetailFetcher.PROP_AREA;
+import static com.mapzen.places.api.internal.PlaceDetailFetcher.PROP_ID;
 import static org.mockito.Matchers.any;
+import static org.mockito.Matchers.anyString;
 import static org.mockito.Matchers.eq;
 import static org.mockito.Mockito.mock;
 import static org.mockito.Mockito.verify;
 import retrofit2.Callback;
 
 public class PeliasPlaceDetailFetcherTest {
 
-  Pelias pelias;
-  PeliasPlaceDetailFetcher fetcher;
+  private Pelias pelias;
+  private PeliasPlaceDetailFetcher fetcher;
 
-  @Before
-  public void setup() {
+  @Before public void setup() {
     pelias = mock(Pelias.class);
     PeliasCallbackHandler callbackHandler = mock(PeliasCallbackHandler.class);
     fetcher = new PeliasPlaceDetailFetcher(pelias, callbackHandler);
   }
 
-  @Test
-  public void fetchDetails_coords_shouldCallPelias() throws Exception {
-    LngLat coords = new LngLat(1, 2);
+  @Test public void fetchDetails_props_shouldCallPelias() throws Exception {
     Map props = new HashMap();
-    fetcher.fetchDetails(coords, props, null);
-    verify(pelias).reverse(eq(coords.latitude), eq(coords.longitude), any(Callback.class));
+    fetcher.fetchDetails(props, null);
+    verify(pelias).place(anyString(), any(Callback.class));
   }
 
-  @Test
-  public void fetchDetails_gid_shouldCallPelias() throws Exception {
+  @Test public void fetchDetails_gid_shouldCallPelias() throws Exception {
     String gid = "123";
     fetcher.fetchDetails(gid, null);
     verify(pelias).place(eq(gid), any(Callback.class));
   }
+
+  @Test public void fetchDetails_props_shouldConstructGidWithWayIfHasArea() throws Exception {
+    Map props = new HashMap();
+    props.put(PROP_ID, "123.000");
+    props.put(PROP_AREA, "456.000");
```

Comment:
Also from the above link only, it seems `highway=footway` `closed-ways` are not assumed to have the area property, how do these get distinguished between a node or a way?